pyomo/solvers/plugins/solvers/gurobi_direct.py

```diff
@@ -131,43 +131,29 @@ def _apply_solver(self):
     def _get_expr_from_pyomo_repn(self, repn, max_degree=2):
         referenced_vars = ComponentSet()
 
-        degree = canonical_degree(repn)
+        degree = repn.polynomial_degree()
         if (degree is None) or (degree > max_degree):
             raise DegreeError('GurobiDirect does not support expressions of degree {0}.'.format(degree))
 
-        if isinstance(repn, LinearCanonicalRepn):
-            if (repn.linear is not None) and (len(repn.linear) > 0):
-                list(map(referenced_vars.add, repn.variables))
-                new_expr = self._gurobipy.LinExpr(repn.linear, [self._pyomo_var_to_solver_var_map[i] for i in repn.variables])
```

Comment:
If we know the expression is linear, this is a much faster way to create the gurobi expression. I may take a stab at updating this method.

Reply:
I took a stab at making this change.  Can you double-check this for me?

Reply:
This looks good now. Thanks.